### PR TITLE
Added docker hardening config

### PR DESCRIPTION
**Pull request by barchen1**
## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/20951

## Does it break backward compatibility?
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/configure_and_test_integration_instances.py [hunk 1]
```diff
@@ -649,14 +706,10 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
 
     if status_code >= 300 or status_code < 200:
         message = result_object.get('message', '')
-        msg = "Failed to set GCP bucket server config - with status code " + str(status_code) + '\n' + message
+        msg = error_msg + str(status_code) + '\n' + message
```
Comment (yaakovi) [inline Tests/configure_and_test_integration_instances.py:709]: use f-string
```suggestion
        msg = f'{error_msg} {status_code}\n{message}'
```

File: Tests/configure_and_test_integration_instances.py [hunk 2]
```diff
@@ -626,16 +626,73 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
...
+        'content.pack.verify': 'false',
+        'marketplace.initial.sync.delay': '0',
+        'content.pack.ignore.missing.warnings.contentpack': 'true',
+        'marketplace.bootstrap.bypass.url':
+            'https://storage.googleapis.com/marketplace-ci-build/content/builds/{}/{}'.format(
+                branch_name, ci_build_number)
+
+    }
+    error_msg = "Failed to set GCP bucket server config - with status code "
+    response_data, status_code, _ = update_server_configuration(client, server_configuration, error_msg)
```
Comment (yaakovi) [inline Tests/configure_and_test_integration_instances.py:639]:
```suggestion
    return update_server_configuration(client, server_configuration, error_msg)
```

File: Tests/configure_and_test_integration_instances.py [hunk 3]
```diff
@@ -626,16 +626,73 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
...
+    # make request to update server configs
+    server_configuration = {
+        'docker.cpu.limit': '1.0',
+        'docker.run.internal.asuser': 'true',
+        'limit.docker.cpu': 'true',
+        'python.pass.extra.keys': '--memory=1g##--memory-swap=-1##--pids-limit=256##--ulimit=nofile=1024:8192'
+    }
+    error_msg = "Failed to set docker hardening server config - with status code "
+
+    response_data, status_code, _ = update_server_configuration(client, server_configuration, error_msg)
```
Comment (yaakovi) [inline Tests/configure_and_test_integration_instances.py:666]:
```suggestion
    return update_server_configuration(client, server_configuration, error_msg)
```

File: Tests/configure_and_test_integration_instances.py [hunk 4]
```diff
@@ -626,16 +626,73 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
...
+
+
+def get_pack_ids_to_install():
+    with open('./Tests/content_packs_to_install.txt', 'r') as packs_stream:
+        pack_ids = packs_stream.readlines()
+        return [pack_id.rstrip('\n') for pack_id in pack_ids]
```
Comment (yaakovi) [inline Tests/configure_and_test_integration_instances.py:672]: move back to the original location (above `main`, below `update_server_configuration`)

File: Tests/configure_and_test_integration_instances.py [hunk 5]
```diff
@@ -649,14 +706,10 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
...
+        msg = error_msg + str(status_code) + '\n' + message
         print_error(msg)
-
-
-def get_pack_ids_to_install():
-    with open('./Tests/content_packs_to_install.txt', 'r') as packs_stream:
-        pack_ids = packs_stream.readlines()
-        return [pack_id.rstrip('\n') for pack_id in pack_ids]
+    # client.api_client.pool.close()
+    return response_data, status_code, _
```
Comment (yaakovi) [inline Tests/configure_and_test_integration_instances.py:712]:
```suggestion
    return response_data, status_code
```

File: Tests/configure_and_test_integration_instances.py [hunk 6]
```diff
@@ -626,16 +626,73 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
...
+
+def set_docker_hardening_for_build(client, prints_manager):
+    """Sets docker hardening configuration
+
+    Args:
+        client (demisto_client): The configured client to use.
+        prints_manager (ParallelPrintsManager): Print manager object
+
+    Returns:
+        None
```
Comment (yaakovi) [inline Tests/configure_and_test_integration_instances.py:650]: update

File: Tests/configure_and_test_integration_instances.py (hunk 2 above)
Comment (barchen1) [reply Tests/configure_and_test_integration_instances.py]: done

File: Tests/configure_and_test_integration_instances.py (hunk 6 above)
Comment (barchen1) [reply Tests/configure_and_test_integration_instances.py]: done

File: Tests/configure_and_test_integration_instances.py (hunk 3 above)
Comment (barchen1) [reply Tests/configure_and_test_integration_instances.py]: done

File: Tests/configure_and_test_integration_instances.py (hunk 1 above)
Comment (barchen1) [reply Tests/configure_and_test_integration_instances.py]: done

File: Tests/configure_and_test_integration_instances.py (hunk 5 above)
Comment (barchen1) [reply Tests/configure_and_test_integration_instances.py]: done

File: Tests/configure_and_test_integration_instances.py (hunk 4 above)
Comment (barchen1) [reply Tests/configure_and_test_integration_instances.py]: done

File: Tests/configure_and_test_integration_instances.py [hunk 7]
```diff
@@ -722,8 +775,10 @@ def set_marketplace_gcp_bucket_for_build(client, prints_manager, branch_name, ci
 
     if status_code >= 300 or status_code < 200:
         message = result_object.get('message', '')
-        msg = "Failed to set GCP bucket server config - with status code " + str(status_code) + '\n' + message
+        msg = f'{error_msg} {status_code}\n{message}'
         print_error(msg)
+    # client.api_client.pool.close()
```
Comment (yaakovi) [inline Tests/configure_and_test_integration_instances.py:780]: ?

Comment (barchen1) [reply Tests/configure_and_test_integration_instances.py]: deleted

**Pull request merged**